rl_algorithms/common/abstract/learner.py

```diff
@@ -0,0 +1,67 @@
+from abc import ABC, abstractmethod
+import argparse
+import os
+from typing import Tuple, Union
+
+import torch
+
+from rl_algorithms.utils.config import ConfigDict
+
+TensorTuple = Tuple[torch.Tensor, ...]
+
+
+class Learner(ABC):
+    """Abstract learner for all learners
+
+    Attributes:
+        args (argparse.Namespace): arguments including hyperparameters and training settings
+        hyper_params (ConfigDict): hyper-parameters
+        log_cfg (ConfigDict): configuration for saving log and checkpoint
```

Comment:
It doesn't matter that it is removed if it doesn't use `log_cfg`.

Reply:
Learner needs log_cfg anyways if ckpt_path and sha declarations are moved to learner abstract class.